services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5526,17 +5525,17 @@ public boolean isLeaderControllerFor(String clusterName) {
   }
 
   /**
-   * Calculate number of partition for given store by give size.
+   * Calculate number of partition for given store.
    */
   @Override
-  public int calculateNumberOfPartitions(String clusterName, String storeName, long storeSize) {
+  public int calculateNumberOfPartitions(String clusterName, String storeName) {
     checkControllerLeadershipFor(clusterName);
     HelixVeniceClusterResources resources = getHelixVeniceClusterResources(clusterName);
     Store store = resources.getStoreMetadataRepository().getStoreOrThrow(storeName);
     VeniceControllerClusterConfig config = resources.getConfig();
     return PartitionUtils.calculatePartitionCount(
```

Comment:
Can we short the method by checking the store partition count here and return early? Also validate the store is indeed hybrid when partition count is set, else recalculate.

Reply:
1. PartitionUtils.calculatePartitionCount already returns store partition count when it is set. IMHO, no need to duplicate the logic.
2. Even if it is not hybrid store, store-level partition count could also be set if SREs or owners specifically update the store with a partition number. IMHO, we should honor store-level partition count rather than recalculate.